packages/store/src/usersSlice/thunks.ts

```diff
@@ -377,33 +462,42 @@ const customPermissionHandler = (accu, permission) => {
   };
 };
 
-const mapPermissionSet = (permissionSetName, names, scope, existingGroupsPermissions = {}) => {
-  const permission = Object.values(uiPermissionsById).find(permission => permission.permissionSets[scope] === permissionSetName).value;
+const mapPermissionSet = (
+  permissionSetName: PermissionSetIds,
+  names: string[],
+  scope: UiPermissionsByAreaKeys,
+  existingGroupsPermissions: CombinedPermission = {}
+) => {
+  const permission = Object.values(uiPermissionsById).find(permission => permission.permissionSets[scope] === permissionSetName)?.value as UiPermissionsByIdKey;
   const scopedPermissions = names.reduce((accu, name) => combinePermissions(accu, { [name]: [permission] }), existingGroupsPermissions);
   return Object.entries(scopedPermissions).reduce((accu, [key, permissions]) => ({ ...accu, [key]: deriveImpliedAreaPermissions(scope, permissions) }), {});
 };
 
-const isEmptyPermissionSet = permissionSet =>
+const isEmptyPermissionSet = (permissionSet: Partial<UiPermissions>) =>
   !Object.values(permissionSet).reduce((accu, permissions) => {
     if (Array.isArray(permissions)) {
       return accu || !!permissions.length;
     }
     return accu || !isEmpty(permissions);
   }, false);
 
-const parseRolePermissions = ({ permission_sets_with_scope = [], permissions = [] }, permissionSets) => {
+//TODO: remove `permissions` if that's legacy
+const parseRolePermissions = (
+  { permission_sets_with_scope = [], permissions = [] }: ApiRole & { permissions?: any[] },
+  permissionSets: Record<string, PermissionSet>
+) => {
   const preliminaryResult = permission_sets_with_scope.reduce(
     (accu, permissionSet) => {
-      let processor = permissionSets[permissionSet.name];
+      const processor = permissionSets[permissionSet.name];
       if (!processor) {
         return accu;
       }
-      const scope = Object.keys(scopedPermissionAreas).find(scope => uiPermissionsByArea[scope].scope === permissionSet.scope?.type);
+      const scope = Object.keys(scopedPermissionAreas).find(scope => uiPermissionsByArea[scope].scope === permissionSet.scope?.type) as UiPermissionsByAreaKeys;
       if (scope) {
-        const result = mapPermissionSet(permissionSet.name, permissionSet.scope.value, scope, accu.uiPermissions[scope]);
+        const result = mapPermissionSet(permissionSet.name, permissionSet.scope?.value || [], scope, accu.uiPermissions[scope] as CombinedPermission);
```

Comment:
```suggestion
        const result = mapPermissionSet(permissionSet.name, permissionSet.scope.value, scope, accu.uiPermissions[scope] as CombinedPermission);
```
with `scope` truthy right above, `value` should be required - no?

Reply:
ah, so I guess: 
```suggestion
        const result = mapPermissionSet(permissionSet.name, permissionSet.scope!.value, scope, accu.uiPermissions[scope] as CombinedPermission);
```
should clarify this for ts instead.